youtube_dl/downloader/dash.py

```diff
@@ -40,7 +40,8 @@ def combine_url(base_url, target_url):
         fragment_retries = self.params.get('fragment_retries', 0)
         skip_unavailable_fragments = self.params.get('skip_unavailable_fragments', True)
 
-        def append_url_to_file(target_url, tmp_filename, segment_name):
+        def append_url_to_file(tmp_filename, segment, skip_unavailable_fragments):
```

Comment:
i have proposed to add it as a method in FragmentFD in https://github.com/rg3/youtube-dl/pull/10451#issuecomment-242914101(already moved part of the code in https://github.com/rg3/youtube-dl/pull/9937), but as @dstftw the code for fragment skipping and retries can be moved when it's ready to be used by all the fragment downloaders.